### PR TITLE
SQLite: Add named parameters support

### PR DESCRIPTION
**Pull request by atishay**
### Brief summary of the change made
Added support for SQLite named parameters as per https://www.sqlite.org/lang_expr.html#varparam
https://www.sqlite.org/c3ref/bind_blob.html

Inspired by https://github.com/sqlfluff/sqlfluff/blob/main/src/sqlfluff/dialects/dialect_bigquery.py#L2214

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

**Comment by atishay** [comment]
I can merge the multiple PRs into one if that helps. I started with one feature request and as I went through my code found more parser problems that I raised separately.

File: src/sqlfluff/dialects/dialect_sqlite.py [hunk 1]
```diff
@@ -89,6 +91,36 @@
...
+            "question_literal",
+            r"\?[0-9]+",
+            LiteralSegment,
+            segment_kwargs={"type": "question_literal"},
+        ),
+        RegexLexer(
+            "dollar_literal",
+            r"\$[a-zA-Z0-9_]+",
+            LiteralSegment,
+            segment_kwargs={"type": "question_literal"},
```
Comment (WittierDinosaur) [inline src/sqlfluff/dialects/dialect_sqlite.py:118]:
```suggestion
            segment_kwargs={"type": "dollar_literal"},
```

Comment (atishay) [reply src/sqlfluff/dialects/dialect_sqlite.py:118]: Will do.

**Review by greg-finley — APPROVED** [review]
Awesome, thanks!

**Pull request merged**